instrumentation/openai/openai-java-1.1/library/src/main/java/io/opentelemetry/instrumentation/openai/v1_1/ChatCompletionEventsHelper.java

```diff
@@ -215,21 +220,247 @@ private static LogRecordBuilder newEvent(Logger eventLogger, String name) {
   private static Value<?> buildToolCallEventObject(
       ChatCompletionMessageToolCall call, boolean captureMessageContent) {
     Map<String, Value<?>> result = new HashMap<>();
-    result.put("id", Value.of(call.id()));
-    result.put("type", Value.of("function")); // "function" is the only currently supported type
-    result.put("function", buildFunctionEventObject(call.function(), captureMessageContent));
+    FunctionAccess functionAccess = getFunctionAccess(call);
+    if (functionAccess != null) {
+      result.put("id", toValue(functionAccess.id()));
+      result.put("type", Value.of("function")); // "function" is the only currently supported type
+      result.put("function", buildFunctionEventObject(functionAccess, captureMessageContent));
+    }
     return Value.of(result);
   }
 
   private static Value<?> buildFunctionEventObject(
-      ChatCompletionMessageToolCall.Function function, boolean captureMessageContent) {
+      FunctionAccess functionAccess, boolean captureMessageContent) {
     Map<String, Value<?>> result = new HashMap<>();
-    result.put("name", Value.of(function.name()));
+    result.put("name", toValue(functionAccess.name()));
     if (captureMessageContent) {
-      result.put("arguments", Value.of(function.arguments()));
+      result.put("arguments", toValue(functionAccess.arguments()));
     }
     return Value.of(result);
   }
 
+  @Nullable
+  private static Value<String> toValue(@Nullable String string) {
```

Comment:
IIUC, we have null handling here because of the very rare situation where reflection fails for some reason - this is despite also having nice `isAvailable` checks.

Any thoughts on having `invokeStringHandle` return `""` in the failure case since we expect it to basically not happen?

Otherwise, it's probably worth adding a comment summarizing the above to clarify this is an issue with the code here and not the API responses